### PR TITLE
Add auction app

### PR DESCRIPTION
**Pull request by nikitayutanov**

### CODE REVIEW

File: apps/auction/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+module.exports = {
+  env: {
```
Comment (osipov-mit) [inline apps/auction/.eslintrc.js:2]:
```suggestion
module.exports = {
  root: true,
  env: {
```

File: apps/auction/package.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    "@types/react": "^18.0.15",
+    "@types/react-dom": "^18.0.6",
+    "clsx": "^1.2.1",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "react-router-dom": "^6.3.0",
+    "react-scripts": "^5.0.1",
+    "react-transition-group": "4.4.2",
+    "sass": "^1.54.0",
+    "typescript": "^4.7.4",
```
Comment (Pinkdesu) [inline apps/auction/package.json:30]: it's dev dep, same as @types

File: apps/auction/src/hooks/api.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { useReadState, useSendMessage } from '@gear-js/react-hooks';
+import { useMemo } from 'react';
+import auctionMetaWasm from 'assets/wasm/dutch_auction.meta.wasm';
+import nftMetaWasm from 'assets/wasm/nft.meta.wasm';
+import { ADDRESS } from 'consts';
+import { AuctionState, NFTState } from 'types';
+import { Hex } from '@gear-js/api';
+
+function useAuction() {
+  const payload = useMemo(() => ({ Info: null }), []);
```
Comment (Pinkdesu) [inline apps/auction/src/hooks/api.ts:10]: maybe move to constants

File: apps/auction/src/pages/home/Home.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+function Home() {
+  const auction = useAuction();
+  const { nftContractActorId, tokenId } = auction || {};
+
+  const { nft, isNftStateRead } = useNft(nftContractActorId, tokenId);
+  const { media } = nft || {};
+  const src = `${ADDRESS.IPFS_GATEWAY_ADDRESS}/${media}`;
+
+  const { status, resetStatus, setExpiredStatus, setRunningStatus } = useStatus(auction?.status);
+  const purchasePrice = typeof status === 'object' ? status.Purchased.price : undefined;
```
Comment (Pinkdesu) [inline apps/auction/src/pages/home/Home.tsx:17]: status?.Purchased.price

File: apps/auction/src/pages/home/Home.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  return auction && isNftStateRead ? (
+    <>
+      {status === 'IsRunning' && (
```
Comment (Pinkdesu) [inline apps/auction/src/pages/home/Home.tsx:21]: move statuses to contacts, cuz you use it in different places

File: apps/auction/src/pages/home/sell/form/Form.tsx [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    const hourSeconds = +hours * MULTIPLIER.MINUTES * MULTIPLIER.SECONDS;
+    const minuteSeconds = +minutes * MULTIPLIER.SECONDS;
+
+    return hourSeconds + minuteSeconds + +seconds;
+  };
+
+  const price = +values.startingPrice - getSeconds() * +values.discountRate;
+  const priceClassName = clsx(styles.price, price > 0 && styles.success, price < 0 && styles.error);
+
+  const isNftContractAddressValid = async () => api.program.exists(nftContractActorId as Hex);
```
Comment (Pinkdesu) [inline apps/auction/src/pages/home/sell/form/Form.tsx:54]: remove the 'async'

File: apps/auction/src/pages/home/sell/form/Form.tsx [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+  const getSeconds = () => {
+    const hourSeconds = +hours * MULTIPLIER.MINUTES * MULTIPLIER.SECONDS;
+    const minuteSeconds = +minutes * MULTIPLIER.SECONDS;
+
+    return hourSeconds + minuteSeconds + +seconds;
+  };
+
+  const price = +values.startingPrice - getSeconds() * +values.discountRate;
+  const priceClassName = clsx(styles.price, price > 0 && styles.success, price < 0 && styles.error);
```
Comment (Pinkdesu) [inline apps/auction/src/pages/home/sell/form/Form.tsx:52]: price > 0 ? styles.success : styles.error

**Review by aderrod94 — APPROVED** [review]

File: apps/auction/package.json (hunk 2 above)
Comment (nikitayutanov) [reply apps/auction/package.json:30]: [source](https://stackoverflow.com/a/44872787)

File: apps/auction/src/hooks/api.ts (hunk 3 above)
Comment (nikitayutanov) [reply apps/auction/src/hooks/api.ts:10]: I prefer to not separate small encapsulated values

File: apps/auction/src/pages/home/Home.tsx (hunk 4 above)
Comment (nikitayutanov) [reply apps/auction/src/pages/home/Home.tsx:17]: if `status` is object, these props gonna be there for sure

File: apps/auction/src/pages/home/sell/form/Form.tsx (hunk 7 above)
Comment (nikitayutanov) [reply apps/auction/src/pages/home/sell/form/Form.tsx:52]: don't need color if `0`

File: apps/auction/package.json (hunk 2 above)
Comment (Pinkdesu) [reply apps/auction/package.json:30]: Information outdated, do not agree

File: apps/auction/src/pages/home/Home.tsx (hunk 4 above)
Comment (Pinkdesu) [reply apps/auction/src/pages/home/Home.tsx:17]: use es6 syntax

**Review by Pinkdesu — APPROVED** [review]

**Pull request merged**